src/platform_impl/linux/wayland/event_loop/mod.rs

```diff
@@ -403,16 +419,17 @@ impl<T: 'static> EventLoop<T> {
             // woken up by messages arriving from the Wayland socket, to avoid delaying the
             // dispatch of these events until we're woken up again.
             let instant_wakeup = {
-                let handle = self.event_loop.handle();
-                let source = self.wayland_source.clone();
-                let dispatched = handle.with_source(&source, |wayland_source| {
-                    let queue = wayland_source.queue();
-                    self.with_state(|state| {
-                        queue.dispatch_pending(state, |_, _, _| unimplemented!())
-                    })
-                });
+                let mut wayland_source = self.wayland_dispatcher.as_source_mut();
+                let queue = wayland_source.queue();
+                let state = match &mut self.window_target.p {
```

Comment:
Fortunately you haven't introduced it in this PR, but I have to say `p` isn't the greatest field name.

Reply:
I know, but it's pretty old name in `event_loop.rs` and any refactoring to such names should be done in separate PRs, since it's touching the entire winit source tree.

Reply:
Definitely, just had to point this out.